release/Magarena/scripts/Balustrade_Spy.groovy

```diff
@@ -0,0 +1,29 @@
+// When SN enters the battlefield, target player reveals cards from the top of his or her library until he or she reveals a land card, then puts those cards into his or her graveyard.
+
+[
+    new EntersBattlefieldTrigger() {
+        @Override
+        public MagicEvent executeTrigger(final MagicGame game,final MagicPermanent permanent, final MagicPayedCost payedCost) {
+            return new MagicEvent(
+                permanent,
+                NEG_TARGET_PLAYER,
+                this,
+                "Target player\$ reveals cards from the top of his or her library until he or she reveals a land card, " +
+                "then puts all cards revealed this way into his or her graveyard."
+            );
+        }
+        @Override
+        public void executeEvent(final MagicGame game, final MagicEvent event) {
+            event.processTargetPlayer(game, {
+                final MagicCardList library = it.getLibrary();
+                int landCards = 0;
```

Comment:
based on the wording, this should be done as two steps. First gather a list of cards up to the first land cards, then use RevealAction to reveal them at once, and then ShiftCardAction to move them to the graveyard. Use of MillLibraryAction would generate a lot of log messages, one per mill action.

Reply:
I have inspired in Mind Grind and Mirko Vosk groovy code, which are doing it the same way (revealing card one by one and putting it to graveyard until conditions are met - which may be a bit like how actual players will resolve the effect in real game :). So I think when fixing this, I'll fix those two too to use one single mill action.

Reply:
I have generalized the MillLibraryAction into MillLibraryAction (mill fixed amount) and MillLibraryUntilAction (mill until certain amount of some type of cards are seen).

I think it might be worth trying to add it into MagicRuleEventAction, as that could make perhaps these 3 cards work even without groovy and some other non-implemented cards might start to just work with that (Undercity Informer, Mind Funeral, Destroy the Evidence, Consuming Aberration?). I'll try that in the next merge request.